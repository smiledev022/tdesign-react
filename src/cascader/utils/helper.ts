import { TreeNode } from '../interface';

/**
 * 获取node节点的全路径
 * @param node
 * @returns
 */
export function getFullPathLabel(node: TreeNode) {
  return node
    .getPath()
    .map((node: TreeNode) => node.label)
    .join('/');
}
