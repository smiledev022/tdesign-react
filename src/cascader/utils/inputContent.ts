import { TreeNode, CascaderContextType, TreeNodeValue } from '../interface';

/**
 * icon Class
 * @param prefix
 * @param CLASSNAMES
 * @param cascaderContext
 * @returns
 */
export function getIconClass(prefix: string, CLASSNAMES: any, cascaderContext: CascaderContextType) {
  const { visible } = cascaderContext;
  return [
    `${prefix}-cascader-icon`,
    {
      [CLASSNAMES.STATUS.visible]: visible,
    },
  ];
}

/**
 * Cascader外显div class
 * @param prefix
 * @param CLASSNAMES
 * @param cascaderContext
 * @returns
 */
export function getCascaderInnerClasses(prefix: string, CLASSNAMES: any, cascaderContext: CascaderContextType) {
  const { disabled, visible, size, multiple } = cascaderContext;
  return [
    `${prefix}-cascader`,
    {
      [CLASSNAMES.STATUS.disabled]: disabled,
      [CLASSNAMES.STATUS.active]: visible,
      [CLASSNAMES.SIZE[size]]: size,
      [`${name}-is-multiple`]: multiple,
    },
  ];
}

/**
 * closeIcon 是否显示
 * @param isHover
 * @param cascaderContext
 * @returns
 */
export function getCloseShow(isHover: boolean, cascaderContext: CascaderContextType) {
  const { multiple, model, disabled, clearable, visible } = cascaderContext;
  return !!(
    !visible &&
    clearable &&
    isHover &&
    !disabled &&
    ((!multiple && model) || (multiple && (model as TreeNodeValue[]).length))
  );
}

/**
 * Placeholder 是否显示
 * @param isHover
 * @param cascaderContext
 * @returns
 */
export function getPlaceholderShow(
  cascaderContext: CascaderContextType,
  singleContent: string,
  multipleContent: TreeNode[],
) {
  const { filterable, visible, multiple } = cascaderContext;
  const valEmpty = !(multiple ? multipleContent.length : singleContent);
  if (filterable) {
    return valEmpty && !visible;
  }
  return valEmpty;
}

/**
 * 单选状态下内容
 * @param isHover
 * @param cascaderContext
 * @returns
 */
export function getSingleContent(cascaderContext: CascaderContextType) {
  const { model, multiple, treeStore, showAllLevels } = cascaderContext;
  if (multiple || !model) return;
  const node = treeStore && treeStore.getNodes(model as TreeNodeValue | TreeNode);
  if (!(node && node.length)) {
    return '';
  }
  const path = node && node[0].getPath();
  if (path && path.length) {
    return showAllLevels ? path.map((node: TreeNode) => node.label).join('/') : path[path.length - 1].label;
  }
  return model as string;
}

/**
 * 多选状态下选中内容
 * @param isHover
 * @param cascaderContext
 * @returns
 */
export function getMultipleContent(cascaderContext: CascaderContextType) {
  const { model, multiple, treeStore } = cascaderContext;

  if (!multiple || !model) return [];
  const node = treeStore && treeStore.getNodes(model as TreeNodeValue | TreeNode);
  if (!node) return [];
  const path = (model as TreeNodeValue[]).map((item: TreeNodeValue) => {
    const node = treeStore.getNodes(item);
    return node[0];
  });
  if (model && (model as TreeNodeValue[]).length) {
    return path && path.length ? path : (model as TreeNode[]);
  }
  return [];
}

/**
 * 点击popup outside副作用
 * @param ref
 * @param cascaderContext
 * @param event
 * @returns
 */
export function outerClickListenerEffect(
  ref: HTMLElement,
  cascaderContext: CascaderContextType,
  event: MouseEvent | TouchEvent,
) {
  const { visible, setVisible } = cascaderContext;
  if (!ref || ref.contains(event.target as Node)) {
    return;
  }

  if (visible) {
    setVisible(false);
  }
}

/**
 * closeIcon点击副作用
 * @param cascaderContext
 */
export function closeIconClickEffect(cascaderContext: CascaderContextType) {
  const { setVisible, setModel, treeStore, multiple } = cascaderContext;

  setModel(multiple ? [] : '');
  treeStore.resetChecked();
  treeStore.resetExpanded();
  setVisible(false);
}

/**
 * tag 关闭按钮点击副作用
 * @param cascaderContext
 */
export function handleRemoveTagEffect(cascaderContext: CascaderContextType, node: TreeNode) {
  const { disabled, setModel } = cascaderContext;

  if (disabled) return;
  const checked = node.setChecked(!node.isChecked());
  setModel(checked);
}

/**
 * 外显content点击副作用
 * @param cascaderContext
 */
export function innerContentClickEffect(cascaderContext: CascaderContextType) {
  const { setVisible, visible, filterActive, disabled } = cascaderContext;

  if (!disabled) {
    if (visible && filterActive) return; // input filter is active
    setVisible(!visible);
  }
}
