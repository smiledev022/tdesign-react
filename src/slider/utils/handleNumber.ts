export const numberToPencent = (number: number) => `${number * 100}%`;
