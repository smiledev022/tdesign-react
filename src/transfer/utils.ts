import isEmpty from 'lodash/isEmpty';
import isFunction from 'lodash/isFunction';
import isString from 'lodash/isString';
import React from 'react';
import { TNode } from '../common';

export const getDefaultValue = <T>(value: T | T[], defaultValue?: T): T[] => {
  if (isEmpty(value)) {
    return Array.isArray(defaultValue) ? defaultValue : [defaultValue, defaultValue];
  }

  if (Array.isArray(value)) return value;

  return [defaultValue, defaultValue];
};

export const getJSX = (value: string | TNode) => {
  if (isString(value)) return value;

  if (isFunction(value)) return value();

  if (React.isValidElement(value)) return value;

  return value;
};
