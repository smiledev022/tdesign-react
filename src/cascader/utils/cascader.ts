import { getFullPathLabel } from './helper';
import { TreeNode, TreeNodeValue, TreeOptionData, CascaderContextType } from '../interface';

/**
 * treeValue计算方法
 * @param model
 * @returns
 */
export const getTreeValue = (model: CascaderContextType['model']) => {
  let treeValue: TreeNodeValue[] = [];
  if (Array.isArray(model)) {
    if (model.length > 0 && typeof model[0] === 'object') {
      treeValue = (model as TreeOptionData[]).map((val) => val.value);
    } else if (model.length) {
      treeValue = model as TreeNodeValue[];
    }
  } else if (model) {
    if (typeof model === 'object') {
      treeValue = [(model as TreeOptionData).value];
    } else {
      treeValue = [model];
    }
  }
  return treeValue;
};

/**
 * input和treeStore变化的副作用
 * @param inputVal
 * @param treeStore
 * @param setTreeNodes
 * @returns
 */
export const treeNodesEffect = (
  inputVal: CascaderContextType['inputVal'],
  treeStore: CascaderContextType['treeStore'],
  setTreeNodes: CascaderContextType['setTreeNodes'],
) => {
  if (!treeStore) return;
  let nodes = [];
  if (inputVal) {
    nodes = treeStore.nodes.filter((node: TreeNode) => {
      const fullPathLabel = getFullPathLabel(node);
      return fullPathLabel.indexOf(inputVal) > -1 && node.isLeaf();
    });
  } else {
    nodes = treeStore.getNodes().filter((node: TreeNode) => node.visible);
  }
  setTreeNodes(nodes);
};

/**
 * 初始化展开阶段与展开状态副作用
 * @param treeStore
 * @param treeValue
 * @param expend
 */
export const treeStoreExpendEffect = (
  treeStore: CascaderContextType['treeStore'],
  treeValue: TreeNodeValue[],
  expend: TreeNodeValue[],
) => {
  // init expanded, 无expend状态时设置
  if (Array.isArray(treeValue) && expend.length === 0) {
    const expandedMap = new Map();
    const [val] = treeValue;
    if (val) {
      expandedMap.set(val, true);
      const node = treeStore.getNode(val);
      if (!node) return;
      node.getParents().forEach((tn: TreeNode) => {
        expandedMap.set(tn.value, true);
      });
      const expandedArr = Array.from(expandedMap.keys());
      treeStore.replaceExpanded(expandedArr);
    }
  }
  // 本地维护 expend，更加可控，不需要依赖于 tree 的状态
  if (treeStore.getExpanded() && expend.length) {
    treeStore.replaceExpanded(expend);
  }
  treeStore.refreshNodes();
};
