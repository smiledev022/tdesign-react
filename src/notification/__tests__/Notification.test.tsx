import React from 'react';
import { testExamples, render, fireEvent } from '@test/utils';
import { Notification, IconFont } from '@tencent/tdesign-react';

testExamples(__dirname);

describe('Notification test', () => {
  test('mount and unmount', () => {
    const wrapper = render(<Notification />);

    expect(() => wrapper.unmount()).not.toThrow();
  });

  test('set props', async () => {
    const titleId = 'title';
    const titleText = '标题';
    const contentId = 'content';
    const contentText = '内容';
    const closeBtnId = 'close';
    const closeBtnText = '关闭';
    const footerId = 'footer';

    const wrapper = render(
      <Notification
        title={<div data-testid={titleId}>{titleText}</div>}
        content={<div data-testid={contentId}>{contentText}</div>}
        theme="success"
        icon={<IconFont name="more" />}
        closeBtn={<div data-testid={closeBtnId}>{closeBtnText}</div>}
        footer={
          <div data-testid={footerId} className="t-notification__detail">
            <span className="t-notification__detail--item">确定</span>
          </div>
        }
        onClickCloseBtn={(event, instance) => instance.close()}
      />,
    );

    expect(wrapper.queryByTestId(titleId)).toHaveTextContent(titleText);
    expect(wrapper.queryByTestId(contentId)).toHaveTextContent(contentText);
    expect(wrapper.queryByTestId(closeBtnId)).toHaveTextContent(closeBtnText);
    expect(wrapper.queryByTestId(footerId)).not.toBeNull();
  });

  test('open and close', async () => {
    const notification = await Notification.open();

    expect(document.querySelectorAll('.t-notification').length).toBe(1);

    notification.close();

    expect(document.querySelectorAll('.t-notification').length).toBe(0);

    const notificationPromise = Notification.open();

    await notificationPromise;

    expect(document.querySelectorAll('.t-notification').length).toBe(1);

    await Notification.close(notificationPromise);

    expect(document.querySelectorAll('.t-notification').length).toBe(0);
  });

  test('open with theme', async () => {
    Notification.closeAll();

    await Notification.info();
    await Notification.success();
    await Notification.warning();
    await Notification.error();

    expect(document.querySelectorAll('.t-notification .t-is-info').length).toBe(1);
    expect(document.querySelectorAll('.t-notification .t-is-success').length).toBe(1);
    expect(document.querySelectorAll('.t-notification .t-is-warning').length).toBe(1);
    expect(document.querySelectorAll('.t-notification .t-is-error').length).toBe(1);
  });

  test('open with placement', async () => {
    Notification.closeAll();

    await Notification.open({ placement: 'top-left' });
    await Notification.open({ placement: 'top-right' });
    await Notification.open({ placement: 'bottom-left' });
    await Notification.open({ placement: 'bottom-right' });

    expect(document.querySelectorAll('.t-notification__show--top-left').length).toBe(1);
    expect(document.querySelectorAll('.t-notification__show--top-right').length).toBe(1);
    expect(document.querySelectorAll('.t-notification__show--bottom-left').length).toBe(1);
    expect(document.querySelectorAll('.t-notification__show--bottom-right').length).toBe(1);
  });

  test('auto close', async () => {
    Notification.closeAll();

    await Notification.open({ duration: 3000 });

    expect(document.querySelectorAll('.t-notification').length).toBe(1);

    jest.runAllTimers();

    expect(document.querySelectorAll('.t-notification').length).toBe(0);
  });

  test('click close button', async () => {
    Notification.closeAll();

    await Notification.open({
      closeBtn: <span id="close_button">关闭</span>,
    });

    fireEvent.click(document.querySelector('#close_button'));

    expect(document.querySelectorAll('.t-notification').length).toBe(0);
  });
});
