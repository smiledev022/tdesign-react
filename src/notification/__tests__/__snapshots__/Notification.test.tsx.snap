// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationBase.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-notification"
  >
    <div
      class="t-notification__main"
    >
      <div
        class="t-notification__title__wrap"
      >
        <span
          class="t-notification__title"
        >
          标题名称
        </span>
        <svg
          class="t-icon t-icon-close t-icon-close"
          version="1.1"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M760.9780096 217.6L806.2328448 262.85483392 557.312 511.744 806.2328448 760.6580096 760.9780096 805.9128448 512.064 556.992 263.17483392 805.9128448 217.92 760.6580096 466.816 511.744 217.92 262.85483392 263.17483392 217.6 512.064 466.496 760.9780096 217.6Z"
            fill-opacity=".9"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="t-notification__content"
      >
        这是一条消息通知
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`NotificationClose.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-notification"
  >
    <div
      class="t-notification__main"
    >
      <div
        class="t-notification__title__wrap"
      >
        <span
          class="t-notification__title"
        >
          自定义关闭按钮(false)
        </span>
      </div>
      <div
        class="t-notification__content"
      >
        这是一条消息通知
      </div>
    </div>
  </div>
  <br />
  <div
    class="t-notification"
  >
    <div
      class="t-notification__main"
    >
      <div
        class="t-notification__title__wrap"
      >
        <span
          class="t-notification__title"
        >
          自定义关闭按钮(文字)
        </span>
      </div>
      <div
        class="t-notification__content"
      >
        这是一条消息通知
      </div>
    </div>
  </div>
  <br />
  <br />
  <div
    class="t-notification"
  >
    <div
      class="t-notification__main"
    >
      <div
        class="t-notification__title__wrap"
      >
        <span
          class="t-notification__title"
        >
          自定义关闭按钮
        </span>
        <div>
          <span>
            x
          </span>
        </div>
      </div>
      <div
        class="t-notification__content"
      >
        这是一条消息通知
      </div>
    </div>
  </div>
  <br />
  <button
    class="t-button t-button--theme-default t-button--variant-base"
  >
    处理开关事件
  </button>
  <button
    class="t-button t-button--theme-default t-button--variant-base"
  >
    处理定时关闭事件（3s）
  </button>
  <br />
  <br />
  <br />
</DocumentFragment>
`;

exports[`NotificationContent.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-notification"
  >
    <div
      class="t-notification__main"
    >
      <div
        class="t-notification__title__wrap"
      >
        <span
          class="t-notification__title"
        >
          自定义内容（字符串）
        </span>
        <svg
          class="t-icon t-icon-close t-icon-close"
          version="1.1"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M760.9780096 217.6L806.2328448 262.85483392 557.312 511.744 806.2328448 760.6580096 760.9780096 805.9128448 512.064 556.992 263.17483392 805.9128448 217.92 760.6580096 466.816 511.744 217.92 262.85483392 263.17483392 217.6 512.064 466.496 760.9780096 217.6Z"
            fill-opacity=".9"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="t-notification__content"
      >
        这是一条消息通知
      </div>
    </div>
  </div>
  <br />
  <div
    class="t-notification"
  >
    <div
      class="t-notification__main"
    >
      <div
        class="t-notification__title__wrap"
      >
        <span
          class="t-notification__title"
        >
          自定义内容
        </span>
        <svg
          class="t-icon t-icon-close t-icon-close"
          version="1.1"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M760.9780096 217.6L806.2328448 262.85483392 557.312 511.744 806.2328448 760.6580096 760.9780096 805.9128448 512.064 556.992 263.17483392 805.9128448 217.92 760.6580096 466.816 511.744 217.92 262.85483392 263.17483392 217.6 512.064 466.496 760.9780096 217.6Z"
            fill-opacity=".9"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div>
        这是一条消息通知
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`NotificationFooter.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-notification"
  >
    <div
      class="t-notification__main"
    >
      <div
        class="t-notification__title__wrap"
      >
        <span
          class="t-notification__title"
        >
          自定义底部详情
        </span>
        <svg
          class="t-icon t-icon-close t-icon-close"
          version="1.1"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M760.9780096 217.6L806.2328448 262.85483392 557.312 511.744 806.2328448 760.6580096 760.9780096 805.9128448 512.064 556.992 263.17483392 805.9128448 217.92 760.6580096 466.816 511.744 217.92 262.85483392 263.17483392 217.6 512.064 466.496 760.9780096 217.6Z"
            fill-opacity=".9"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="t-notification__content"
      >
        这是一条消息通知
      </div>
      <div
        class="t-notification__detail"
      >
        <div
          class="t-notification__detail"
        >
          <span
            class="t-notification__detail--item"
          >
            重启
          </span>
          <span
            class="t-notification__detail--item t-is-active"
          >
            查看详情
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`NotificationIcon.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-notification"
  >
    <i
      class="t-icon t-icon-more t-size-m"
    />
    <div
      class="t-notification__main"
    >
      <div
        class="t-notification__title__wrap"
      >
        <span
          class="t-notification__title"
        >
          自定义icon
        </span>
        <svg
          class="t-icon t-icon-close t-icon-close"
          version="1.1"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M760.9780096 217.6L806.2328448 262.85483392 557.312 511.744 806.2328448 760.6580096 760.9780096 805.9128448 512.064 556.992 263.17483392 805.9128448 217.92 760.6580096 466.816 511.744 217.92 262.85483392 263.17483392 217.6 512.064 466.496 760.9780096 217.6Z"
            fill-opacity=".9"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="t-notification__content"
      >
        这是一条消息通知
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`NotificationOffset.jsx 1`] = `
<DocumentFragment>
  <button
    class="t-button t-button--theme-default t-button--variant-base"
  >
    偏移的信息
  </button>
</DocumentFragment>
`;

exports[`NotificationPlacement.jsx 1`] = `
<DocumentFragment>
  <button
    class="t-button t-button--theme-default t-button--variant-base"
  >
    左上角
  </button>
  <button
    class="t-button t-button--theme-default t-button--variant-base"
  >
    右上角
  </button>
  <br />
  <button
    class="t-button t-button--theme-default t-button--variant-base"
  >
    左下角
  </button>
  <button
    class="t-button t-button--theme-default t-button--variant-base"
  >
    右下角
  </button>
</DocumentFragment>
`;

exports[`NotificationSize.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-notification"
  >
    <div
      class="t-notification__main"
    >
      <div
        class="t-notification__title__wrap"
      >
        <span
          class="t-notification__title"
        >
          宽度固定400px
        </span>
        <svg
          class="t-icon t-icon-close t-icon-close"
          version="1.1"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M760.9780096 217.6L806.2328448 262.85483392 557.312 511.744 806.2328448 760.6580096 760.9780096 805.9128448 512.064 556.992 263.17483392 805.9128448 217.92 760.6580096 466.816 511.744 217.92 262.85483392 263.17483392 217.6 512.064 466.496 760.9780096 217.6Z"
            fill-opacity=".9"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="t-notification__content"
      >
        这是一条消息通知
      </div>
    </div>
  </div>
  <br />
  <div
    class="t-notification"
  >
    <div
      class="t-notification__main"
    >
      <div
        class="t-notification__title__wrap"
      >
        <span
          class="t-notification__title"
        >
          文本内容最多支持三行
        </span>
        <svg
          class="t-icon t-icon-close t-icon-close"
          version="1.1"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M760.9780096 217.6L806.2328448 262.85483392 557.312 511.744 806.2328448 760.6580096 760.9780096 805.9128448 512.064 556.992 263.17483392 805.9128448 217.92 760.6580096 466.816 511.744 217.92 262.85483392 263.17483392 217.6 512.064 466.496 760.9780096 217.6Z"
            fill-opacity=".9"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="t-notification__content"
      >
        这是一条内容为*******************************************************************************
        ***************************************************************************************消息通知
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`NotificationToggle.jsx 1`] = `
<DocumentFragment>
  <button
    class="t-button t-button--theme-default t-button--variant-base"
  >
    自由控制关闭时机
  </button>
  <button
    class="t-button t-button--theme-default t-button--variant-base"
  >
    点击打开多个消息
  </button>
  <button
    class="t-button t-button--theme-default t-button--variant-base"
  >
    点击关闭多个消息
  </button>
</DocumentFragment>
`;

exports[`NotificationType.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-notification"
  >
    <div
      class="t-notification__main"
    >
      <div
        class="t-notification__title__wrap"
      >
        <span
          class="t-notification__title"
        >
          标题名称
        </span>
        <svg
          class="t-icon t-icon-close t-icon-close"
          version="1.1"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M760.9780096 217.6L806.2328448 262.85483392 557.312 511.744 806.2328448 760.6580096 760.9780096 805.9128448 512.064 556.992 263.17483392 805.9128448 217.92 760.6580096 466.816 511.744 217.92 262.85483392 263.17483392 217.6 512.064 466.496 760.9780096 217.6Z"
            fill-opacity=".9"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="t-notification__content"
      >
        这是一条消息通知
      </div>
    </div>
  </div>
  <br />
  <div
    class="t-notification"
  >
    <div
      class="t-notification__icon"
    >
      <svg
        class="t-icon t-icon-info-circle-filled t-is-info"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM544 416L416 416 416 480 480 480 480 736 608 736 608 672 544 672 544 416ZM544 288L480 288 480 352 544 352 544 288Z"
          fill-opacity=".9"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <div
      class="t-notification__main"
    >
      <div
        class="t-notification__title__wrap"
      >
        <span
          class="t-notification__title"
        >
          标题名称
        </span>
        <svg
          class="t-icon t-icon-close t-icon-close"
          version="1.1"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M760.9780096 217.6L806.2328448 262.85483392 557.312 511.744 806.2328448 760.6580096 760.9780096 805.9128448 512.064 556.992 263.17483392 805.9128448 217.92 760.6580096 466.816 511.744 217.92 262.85483392 263.17483392 217.6 512.064 466.496 760.9780096 217.6Z"
            fill-opacity=".9"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="t-notification__content"
      >
        这是一条消息通知
      </div>
    </div>
  </div>
  <br />
  <div
    class="t-notification"
  >
    <div
      class="t-notification__icon"
    >
      <svg
        class="t-icon t-icon-info-circle-filled t-is-warning"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM544 416L416 416 416 480 480 480 480 736 608 736 608 672 544 672 544 416ZM544 288L480 288 480 352 544 352 544 288Z"
          fill-opacity=".9"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <div
      class="t-notification__main"
    >
      <div
        class="t-notification__title__wrap"
      >
        <span
          class="t-notification__title"
        >
          标题名称
        </span>
        <svg
          class="t-icon t-icon-close t-icon-close"
          version="1.1"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M760.9780096 217.6L806.2328448 262.85483392 557.312 511.744 806.2328448 760.6580096 760.9780096 805.9128448 512.064 556.992 263.17483392 805.9128448 217.92 760.6580096 466.816 511.744 217.92 262.85483392 263.17483392 217.6 512.064 466.496 760.9780096 217.6Z"
            fill-opacity=".9"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="t-notification__content"
      >
        这是一条消息通知
      </div>
    </div>
  </div>
  <br />
  <div
    class="t-notification"
  >
    <div
      class="t-notification__icon"
    >
      <svg
        class="t-icon t-icon-check-circle-filled t-is-success"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM690.8655168 353.58818368L464.56401024 579.82818368 328.82684416 444.09785152 283.57201024 489.35268544 464.59134592 670.3720192 509.84617984 625.1171872 736.120352 398.8430176 690.8655168 353.58818368Z"
          fill-opacity=".9"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <div
      class="t-notification__main"
    >
      <div
        class="t-notification__title__wrap"
      >
        <span
          class="t-notification__title"
        >
          标题名称
        </span>
        <svg
          class="t-icon t-icon-close t-icon-close"
          version="1.1"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M760.9780096 217.6L806.2328448 262.85483392 557.312 511.744 806.2328448 760.6580096 760.9780096 805.9128448 512.064 556.992 263.17483392 805.9128448 217.92 760.6580096 466.816 511.744 217.92 262.85483392 263.17483392 217.6 512.064 466.496 760.9780096 217.6Z"
            fill-opacity=".9"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="t-notification__content"
      >
        这是一条消息通知
      </div>
    </div>
  </div>
  <br />
  <div
    class="t-notification"
  >
    <div
      class="t-notification__icon"
    >
      <svg
        class="t-icon t-icon-info-circle-filled t-is-error"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M512 96C741.7504576 96 928 282.24954432 928 512 928 741.7504576 741.7504576 928 512 928 282.24954432 928 96 741.7504576 96 512 96 282.24954432 282.24954432 96 512 96ZM544 416L416 416 416 480 480 480 480 736 608 736 608 672 544 672 544 416ZM544 288L480 288 480 352 544 352 544 288Z"
          fill-opacity=".9"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <div
      class="t-notification__main"
    >
      <div
        class="t-notification__title__wrap"
      >
        <span
          class="t-notification__title"
        >
          标题名称
        </span>
        <svg
          class="t-icon t-icon-close t-icon-close"
          version="1.1"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M760.9780096 217.6L806.2328448 262.85483392 557.312 511.744 806.2328448 760.6580096 760.9780096 805.9128448 512.064 556.992 263.17483392 805.9128448 217.92 760.6580096 466.816 511.744 217.92 262.85483392 263.17483392 217.6 512.064 466.496 760.9780096 217.6Z"
            fill-opacity=".9"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="t-notification__content"
      >
        这是一条消息通知
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`NotificationType4Fn.jsx 1`] = `
<DocumentFragment>
  <button
    class="t-button t-button--theme-default t-button--variant-base"
  >
    信息
  </button>
  <button
    class="t-button t-button--theme-default t-button--variant-base"
  >
    成功
  </button>
  <button
    class="t-button t-button--theme-default t-button--variant-base"
  >
    警告
  </button>
  <button
    class="t-button t-button--theme-default t-button--variant-base"
  >
    错误
  </button>
</DocumentFragment>
`;

exports[`base.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-notification"
  >
    <div
      class="t-notification__main"
    >
      <div
        class="t-notification__title__wrap"
      >
        <span
          class="t-notification__title"
        >
          标题名称
        </span>
        <svg
          class="t-icon t-icon-close t-icon-close"
          version="1.1"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M760.9780096 217.6L806.2328448 262.85483392 557.312 511.744 806.2328448 760.6580096 760.9780096 805.9128448 512.064 556.992 263.17483392 805.9128448 217.92 760.6580096 466.816 511.744 217.92 262.85483392 263.17483392 217.6 512.064 466.496 760.9780096 217.6Z"
            fill-opacity=".9"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="t-notification__content"
      >
        这是一条消息通知
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`icon.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-notification"
  >
    <i
      class="t-icon t-icon-more t-size-m"
    />
    <div
      class="t-notification__main"
    >
      <div
        class="t-notification__title__wrap"
      >
        <span
          class="t-notification__title"
        >
          自定义icon
        </span>
        <svg
          class="t-icon t-icon-close t-icon-close"
          version="1.1"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M760.9780096 217.6L806.2328448 262.85483392 557.312 511.744 806.2328448 760.6580096 760.9780096 805.9128448 512.064 556.992 263.17483392 805.9128448 217.92 760.6580096 466.816 511.744 217.92 262.85483392 263.17483392 217.6 512.064 466.496 760.9780096 217.6Z"
            fill-opacity=".9"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="t-notification__content"
      >
        这是一条消息通知
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`operation.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-notification"
  >
    <div
      class="t-notification__main"
    >
      <div
        class="t-notification__title__wrap"
      >
        <span
          class="t-notification__title"
        >
          自定义底部详情
        </span>
        <svg
          class="t-icon t-icon-close t-icon-close"
          version="1.1"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M760.9780096 217.6L806.2328448 262.85483392 557.312 511.744 806.2328448 760.6580096 760.9780096 805.9128448 512.064 556.992 263.17483392 805.9128448 217.92 760.6580096 466.816 511.744 217.92 262.85483392 263.17483392 217.6 512.064 466.496 760.9780096 217.6Z"
            fill-opacity=".9"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="t-notification__content"
      >
        这是一条消息通知
      </div>
      <div
        class="t-notification__detail"
      >
        <div
          class="t-notification__detail"
        >
          <span
            class="t-notification__detail--item"
          >
            重启
          </span>
          <span
            class="t-notification__detail--item t-is-active"
          >
            查看详情
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
